File: Data Structure II/Java/Day 1/ThreeSum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        Set<List<Integer>> res=new HashSet();
        if(nums==null)
            return null;
        for(int i=0;i<nums.length-1;i++){
            int left=i+1;
            int right=nums.length-1;
            while(left<right){
                int sum=nums[i]+nums[left]+nums[right];
                if(sum==0){
                    res.add(Arrays.asList(nums[i],nums[left],nums[right]));
                    left++;
                    right--;
                }else if(sum>0){
                    right--;
                }else{
                    left++;
                }
            }
            
        }
        return new LinkedList(res);
    }
}

File: Data Structure II/python/Day I/three_sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        lst=[]
        size=len(nums)
        nums.sort()
        for i in range(size):
            j=i+1
            k=size-1
            while j<k:
                if((nums[i]+nums[j]+nums[k])==0) and [nums[i],nums[j],nums[k]] not in lst:
                    lst.append([nums[i],nums[j],nums[k]])
                elif (nums[i]+nums[j]+nums[k])<0:
                    j+=1
                else:
                   k-=1
        return lst
                    
                    
        